metal3-templater.go

```diff
@@ -19,56 +20,122 @@ type templater struct {
 	MachineOSImageURL          string
 
 	// Ironic clouds.yaml data
+	AuthType              string
 	BootstrapIronicURL    string
 	BootstrapInspectorURL string
 	ClusterIronicURL      string
 	ClusterInspectorURL   string
+	IronicUser            string
+	IronicPassword        string
+	InspectorUser         string
+	InspectorPassword     string
 }
 
 func main() {
-	var templateData templater
 
-	if len(os.Args) < 7 {
-		fmt.Printf("usage: <prog> TEMPLATE_FILE INTERFACE NETWORK BOOTSTRAP_IP CLUSTER_IP IMAGE_URL\n")
+	flag.NewFlagSet("bootstrap", flag.ExitOnError)
+	flag.NewFlagSet("noauth", flag.ExitOnError)
+	httpBasicCmd := flag.NewFlagSet("http_basic", flag.ExitOnError)
+	ironicCred := httpBasicCmd.String("ironic-basic-auth", "", "ironic credentials <user>:<password>")
+	inspectorCred := httpBasicCmd.String("inspector-basic-auth", "", "inspector crdentials <user>:<password>")
+	if len(os.Args) < 2 {
+		fmt.Printf("Expected 'bootstrap' 'noauth' or 'http_basic' subcommands\n")
 		os.Exit(1)
 	}
 
-	templateFile := os.Args[1]
+	var auth = os.Args[1]
+	var templateData templater
+	cmd := make([]string, 6)
 
-	templateData.ProvisioningInterface = os.Args[2]
+	switch auth {
+	case "bootstrap":
+		if len(os.Args) < 4 {
```

Comment:
Is there some reason not to define flags for all of the inputs to avoid checking the length of the arguments?

Reply:
Tbh I didn't think about having flag for each argument, I will do some tests

Reply:
@dhellmann I've updated adding arguments to each parameter we have.